src/systemtest/java/reposense/util/SystemTestUtil.java

```diff
@@ -0,0 +1,48 @@
+package reposense.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * Contains utility methods for system tests.
+ */
+public class SystemTestUtil {
+
+    /**
+     * Verifies that all JSON files in the {@code actualDirectory} matches those at the {@code expectedDirectory}.
+     */
+    public static void verifyAllJson(Path expectedDirectory, Path actualDirectory) {
+        try (Stream<Path> pathStream = Files.list(expectedDirectory)) {
+            for (Path file : pathStream.collect(Collectors.toList())) {
+                Path expectedFilePath = expectedDirectory.resolve(file);
+                Path actualFilePath = actualDirectory.resolve(file);
+                if (Files.isDirectory(file)) {
+                    verifyAllJson(expectedFilePath, actualFilePath);
+                }
+                if (file.toString().endsWith(".json")) {
+                    assertJson(expectedFilePath, actualFilePath);
+                }
```

Comment:
Are these two conditions mutually exclusive? Is there a chance where we have a directory that ends with ".json"?

Reply:
![image](https://user-images.githubusercontent.com/65345505/162563542-633b035a-64be-4b36-ae49-fe59f6186c66.png)

Seems like you were right! I've adjusted it to be `else if`.